Hi @kekxv 

Thank you for providing a solution to your problem, unfortunately I don't agree with it, here's why:

```bash
cd lvgl
rm -rf tests docs demos examples
cmake -B build
-- 
-- Using lv_conf.h from the top-level project directory
-- Enabling the building of ThorVG internal
-- Enabling the building of examples
-- Enabling the building of demos
-- Configuring done (0.1s)
CMake Error at env_support/cmake/os_desktop.cmake:238 (add_library):
  No SOURCES given to target: lvgl_examples
Call Stack (most recent call first):
  CMakeLists.txt:26 (include)


CMake Error at env_support/cmake/os_desktop.cmake:260 (add_library):
  No SOURCES given to target: lvgl_demos
Call Stack (most recent call first):
  CMakeLists.txt:26 (include)


CMake Generate step failed.  Build files cannot be regenerated correctly.
``` 

This types of errors will only cause confusion for people IMO as they won't understand why the build fails for them

Maybe other maintainers have a different opinion from mine? cc @kisvegabor @liamHowatt @uLipe @FASTSHIFT @XuNeo

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -34,20 +34,29 @@ jobs:
       # --- New Step: Remove large folders and create a lite zip ---
       - name: Prepare Lite Package
         run: |
-          echo "Removing large directories..."
-          # Remove specific directories shown in your screenshot
-          rm -rf tests
-          rm -rf demos
-          rm -rf scripts
-          rm -rf docs
-          rm -rf examples
+          echo "Preparing lite package in a temporary directory..."
+          
+          # Create a temporary directory for the lite package contents
+          mkdir -p lite
+          
+          echo "Copying files to lite directory (excluding large/unnecessary folders)..."
+          rsync -av \
+            --exclude='.git' \
+            --exclude='.github' \
+            --exclude='tests' \
+            --exclude='demos' \
+            --exclude='scripts' \
+            --exclude='docs' \
+            --exclude='examples' \
+            ./ lite/
           
           # Define the output filename using the tag (e.g., lvgl-lite-v9.3.0.zip)
           ZIP_NAME="lvgl-lite-${{ github.ref_name }}.zip"
           
-          echo "Zipping remaining files..."
-          # Zip current directory, excluding hidden git files
-          zip -r $ZIP_NAME . -x ".git/*" ".github/*"
+          echo "Zipping lite directory..."
+          # Create the zip from inside the lite directory, outputting to the repository root
+          cd lite
+          zip -r "../$ZIP_NAME" . 
           
           # Export the variable for the next step
           echo "ZIP_NAME=$ZIP_NAME" >> $GITHUB_ENV

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -54,12 +54,8 @@ jobs:
 
       # --- New Step: Upload the lite zip to the release ---
       - name: Upload Lite Asset
-        uses: actions/upload-release-asset@v1
+        uses: softprops/action-gh-release@v2
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         with:
-          # Use the upload URL from the create_release step
-          upload_url: ${{ steps.create_release.outputs.upload_url }}
-          asset_path: ./${{ env.ZIP_NAME }}
-          asset_name: ${{ env.ZIP_NAME }}
-          asset_content_type: application/zip
+          files: ./${{ env.ZIP_NAME }}

